### ISSUE TITLE
no-free-form false positive (v6.8.0)

### ISSUE THREAD

**Issue by Vanav** [ansible/ansible-lint#2573]
Code:
```
- name: Configure locale
  command: localectl set-locale LANG=en_GB.UTF-8
  when: not ansible_check_mode
```
ansible-lint:
```
no-free-form: Avoid using free-form when calling module actions. (command) (warning)
roles/base/tasks/main.yml:470 Task/Handler: Configure locale
```
There was no such issue in v6.7.0.

**Comment by felixfontein** [comment]
This is not a false positive. You are using the free-form version of the `command` module.

The non-free-form version would be
```yaml
- name: Configure locale
  command:
    cmd: localectl set-locale LANG=en_GB.UTF-8
  when: not ansible_check_mode
```
([module docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#parameter-cmd))

**Comment by nre-ableton** [comment]
No, this is a false positive. @Vanav's bug report is legit, and I've also seen it pop up on a number of tasks in my playbooks. The problem occurs when ansible-lint sees a command that contains a `=` character. It will also fail on this task:

```yaml
    - name: Test
      ansible.builtin.command: "test --arg=foo"
      when: true
```

The [documentation](https://ansible-lint.readthedocs.io/rules/no-free-form/) includes example problematic code like this:

```yaml
    - name: Create a placefolder file
      ansible.builtin.command: chdir=/tmp touch foo
```

This makes sense, because `chdir` should be a property of the task. However, `test --arg=foo` is not a task property, nor is `LANG=en_GB.UTF-8`.

**Comment by felixfontein** [comment]
According to the module examples (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#examples) `command: localectl set-locale LANG=en_GB.UTF-8` is **also** free-form (also if it wouldn't contain `=`). The the third example in the documentation. This also coincides with how free-form is used in the ansible-core source code.

Apparently ansible-lint interprets free-form differently, though, as the [documentation](https://ansible-lint.readthedocs.io/rules/no-free-form/#correct-code) shows. So if you uses ansible-lint's definition of free-form, then this is indeed a false positive.

**Comment by Vanav** [comment]
AFAIK, only free-form with multiple arguments on one line is considered as bad style (old style).
Free-form with single argument on one line is still valid.

**Comment by ssbarnea** [comment]
The reality is that all forms are free-form, with or without the equal sign on them. Still, in order to lower the impact on the current users, we are going to update the rule to make it be more precise on shell and command, and only trigger if one of the known arguments are present.

This should ease the transition towards full removal of free-form. See #2586

**Comment by Vanav** [comment]
Ansible recommends full removal of free-form? I have not read about it.

More false positive examples:
```
  - name: Generate dummy certificate
    command: >
      openssl req -new -newkey rsa:2048 -sha256 -x509 -nodes -keyout /etc/ssl/private/dummy.key -out /etc/ssl/certs/dummy.pem -days 36500 -subj '/CN=*'
    when:
    - acme_challenge != 'none'

  - name: Install NVM
    shell: |
      set -o pipefail
      curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | NVM_DIR="/opt/nvm" bash
    args:
      creates: /opt/nvm/nvm.sh

  - name: Get list of non-system users on server
    shell: "getent passwd | awk -F: '$3 >= 1000 {print $1}'"
    register: existing_users
    changed_when: false
```

**Comment by nre-ableton** [comment]
Ansible indeed uses free-form syntax frequently, but I agree that it's not optimal and I would support ansible-lint being stricter about this in the future.

@ssbarnea Thanks for the quick (and pragmatic) fix! :+1:
